buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -583,14 +584,15 @@ private static void setSingleBytes(final AbstractByteBuf buf, final long addr, f
     static void getBytes(AbstractByteBuf buf, long addr, int index, OutputStream out, int length) throws IOException {
         buf.checkIndex(index, length);
         if (length != 0) {
-            ByteBuf tmpBuf = buf.alloc().heapBuffer(length);
```

Comment:
isnt this just a very complicated way of making new byte[] ? if I missed some nuance, please let me know ;)

Reply:
@CodingFabian I think the idea was that if you use the pooled allocator you also have heap memory pooled by default and so if doing it like this can reduce allocations. makes sense ?

Reply:
except that it most likely doesnt work here anyway, but that might change witht the 8k buffer.
will add it back. thx